src/utils/update/beneficial_owners_managing_officers_data_fetch.ts

```diff
@@ -1,10 +1,78 @@
 import { ApplicationData } from "../../model";
+import { Request } from "express";
+import { logger } from "../../utils/logger";
+import { EntityNumberKey } from "../../model/data.types.model";
+import { CompanyPersonsWithSignificantControl } from "@companieshouse/api-sdk-node/dist/services/company-psc/types";
+import { mapToManagingOfficer, mapToManagingOfficerCorporate } from "../../utils/update/managing.officer.mapper";
+import { getCompanyOfficers } from "../../service/company.managing.officer.service";
+import { getCompanyPsc } from "../../service/persons.with.signficant.control.service";
+import { mapPscToBeneficialOwnerGov, mapPscToBeneficialOwnerOther, mapPscToBeneficialOwnerTypeIndividual } from "../../utils/update/psc.to.beneficial.owner.type.mapper";
 
-export const hasFetchedBoAndMoData = (appData: ApplicationData) => appData?.update?.bo_mo_data_fetched ?? false;
+export const retrieveBoAndMoData = async (req: Request, appData: ApplicationData) => {
+  if (!hasFetchedBoAndMoData(appData)) {
+    intialiseAppDataUpdate(appData);
+    if (appData.update) {
+      await retrieveBeneficialOwners(req, appData);
+      await retrieveManagingOfficers(req, appData);
+      appData.update.bo_mo_data_fetched = true;
+    }
+  }
+};
+
+const hasFetchedBoAndMoData = (appData: ApplicationData) => appData?.update?.bo_mo_data_fetched ?? false;
 
-export const setFetchedBoMoData = (appData: ApplicationData) => {
+const intialiseAppDataUpdate = (appData: ApplicationData) => {
```

Comment:
Could you rename this? To be about BoAndMo? initializeBoAndMoUpdateAppData or something?

Reply:
done!